### PR TITLE
Wrap dashboard pages with error boundary

### PR DESCRIPTION
**Pull request by akurinnoy**
### What does this PR do?

Wrapping UD page components with error boundary to handle uncaught exceptions that may happen in runtime in components. 

**Screenshots**

Screenshot 2021-02-10 at 14 56 54

*expanded component stack*
Screenshot 2021-02-10 at 14 57 46

### What issues does this PR fix or reference?

fixes https://github.com/eclipse/che/issues/18758

### CODE REVIEW

File: src/services/keycloak/setup.ts [hunk 1]
```diff
@@ -142,10 +132,6 @@ export class KeycloakSetupService {
       if (e.response.status === 404) {
         return;
       }
-      if (this.isDevelopment) {
```
Comment (olexii4) [inline src/services/keycloak/setup.ts:145]: It won't be caught with ErrorBoundary.  I suggest leaving as is.
These changes do not apply to the target commit.

**Review by olexii4 — APPROVED** [review]
LGTM
Please, fix the feedback.

**Review by JPinkney — APPROVED** [review]

**Comment by sleshchenko** [comment]
When devfile registry returns invalid JSON as index.json file, the browser logs: `Failed to fetch devfiles metadata from registry URL: https://devfile-registry-che.192.168.39.241.nip.io/devfiles/index.json,TypeError: (intermediate value).data.map is not a function` but user sees it as no devfile is present in the devfile registry.
It would be cool to display error message for a user, like the configured devfile registry is invalid.

^ it's the bug that we face previously. Now dashboard behaves better, since previously blank page was loaded.

File: src/services/keycloak/setup.ts (hunk 1 above)
Comment (akurinnoy) [reply src/services/keycloak/setup.ts:145]: The exception gets handled [here](https://github.com/eclipse/che-dashboard/blob/cc2e9fbd672993d9ef853cc2114b549aab4accc9/src/index.tsx#L35) and then the dashboard continues to initialize. I tested it running dashboard in dev mode either locally and in Che. I may miss something though, if you think so please elaborate on this.

**Review by sleshchenko — APPROVED** [review]
The described flow works as expected. I did not find any other corner cases yet but in any case it should be good to go, and we'll handle other errors if any separately

Screenshot_20210211_102804

**Pull request merged**